File: yaf-gadomain/src/main/java/ar/edu/ungs/yamiko/ga/domain/impl/BitSetToIntegerRibosome.java
package ar.edu.ungs.yamiko.ga.domain.impl;

import java.io.Serializable;
import java.util.BitSet;

import ar.edu.ungs.yamiko.ga.domain.Ribosome;
import ar.edu.ungs.yamiko.ga.toolkit.BitsStaticHelper;

/**
 * Ribosoma que traduce BitSet a enteros.
 * @author ricardo
 *
 */
public class BitSetToIntegerRibosome implements Ribosome<BitSet>,Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 5472646010255710816L;
	private int floor;
	
	public BitSetToIntegerRibosome(int floor) {
		super();
		this.floor = floor;
	}

	public Object translate(BitSet allele) {
		return floor+BitsStaticHelper.convertInt(allele);
	}
}

File: yaf-gadomain-scala/src/main/scala/ar/edu/ungs/yamiko/ga/domain/impl/BitSetToIntegerRibosome.scala
package ar.edu.ungs.yamiko.ga.domain.impl

import ar.edu.ungs.yamiko.ga.domain.Ribosome
import scala.collection.mutable.BitSet
import ar.edu.ungs.yamiko.ga.toolkit.BitSetHelper

/**
 * Ribosoma que traduce BitSet a enteros.
 * @author ricardo
 *
 */

@SerialVersionUID(1719L)
class BitSetToIntegerRibosome(floor:Int) extends Ribosome[BitSet]{
  
	override def translate(allele:BitSet):Any = floor+BitSetHelper.toInt(allele)
  	
}